test/UnitTest/Microsoft.AspNet.OData.Test.Shared/ScaleDecimalTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Test.Abstraction;
+using Microsoft.AspNet.OData.Test.Builder.TestModels;
+using Microsoft.OData.Edm;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Xunit;
+using System.Globalization;
+
+namespace Microsoft.AspNet.OData.Test
+{
+    public class ScaleDecimalTest
+    {
+        [Fact]
+        public async Task ScaleOfDecimalProperty_ShouldBeCalculated_BasedOnScale()
+        {
+            const string Uri = "http://localhost/odata/Orders";
+            HttpClient client = GetClient();
+
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, Uri);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json;odata.metadata=full"));
+
+            // Act
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            IEnumerable<Order> orders = await ReadEntitiesFromResponse<Order>(response);
+            foreach (Order order in orders)
+            {
+                Assert.Equal("10.00", order.TotalPrice.ToString(CultureInfo.InvariantCulture));
+            }
+        }
+
+        private static HttpClient GetClient()
+        {
+            var controllers = new[] { typeof(MetadataController), typeof(OrdersController) };
```

Comment:
>typeof(MetadataController), [](start = 38, length = 27)

we can exclude MetadataController if no test for metadata

Reply:
excluded